### ISSUE TITLE
About best_valid.pt missing and training epoch.

### ISSUE THREAD

**Issue by jkwang93** [marcopodda/fragment-based-dgm#1]
Great job~ Thank you for your code. When I tested your code recently, I found that the best_valid.pt model was not generated during the training process. Why? And your default epoch is 10. Is it set to a larger value?

**Comment by marcopodda** [comment]
Hi,
I'll try and answer point by point.

1) since validation is time-consuming (it requires exiting the GPU and generating many samples), validation is performed only when the loss falls below a certain threshold, which you can customize by setting the validate_after parameter in the config (see https://github.com/marcopodda/fragment-based-dgm/blob/1a0da9f927084702be85a9b15a4dccb0c3d88e8b/learner/trainer.py#L191) 
If the validated model is not saved, it means that the training loss has not fallen belowthe threshold yet (in this case, the default is 0.3); once it goes below 0.3, the model should be saved correctly.

2) actually training requires less than 10 epoch. If I recall correctly, the model started overfitting after 4-5 epochs (the right amount should be reported in the paper), so we empirically found that that number of epochs was enough to get good results.

**Comment by jkwang93** [comment]
Thank you very much, my problem have been solved~
